internal/internal.go

```diff
@@ -265,7 +265,7 @@ func (p *envoyExtAuthzGrpcServer) Check(ctx ctx.Context, req *ext_authz.CheckReq
 		"txn":                 result.txnID,
 		"metrics":             result.metrics.All(),
 		"total_decision_time": time.Since(start),
-	}).Info("Returning policy decision.")
+	}).Debug("Returning policy decision.")
```

Comment:
Did you check the improvement from this? I could imagine making this and the debug log below conditional if the field construction is costly.

```
if logrus.debugenabled {
    // emit log
}
```

Reply:
@tsandall Here are the benchmark results:

**With Debug log** 
```bash
$ go test -run=XXX -bench=.
goos: darwin
goarch: amd64
pkg: github.com/open-policy-agent/opa-istio-plugin/internal
BenchmarkCheck-8       10000            248507 ns/op
PASS
ok      github.com/open-policy-agent/opa-istio-plugin/internal  2.576s
```

**Without Debug log** 
```bash
$ go test -run=XXX -bench=.
goos: darwin
goarch: amd64
pkg: github.com/open-policy-agent/opa-istio-plugin/internal
BenchmarkCheck-8     10000            222931 ns/op
PASS
ok      github.com/open-policy-agent/opa-istio-plugin/internal  2.317s
```

So the improvement is around 25 microseconds.

Reply:
Added a check to see if debug logging is enabled.